Automattic/wp-calypso
# Replace search component with one from sites page

Related to SELFSERVE-629

Original request from Bart:
![image-2023-06-19-21-49-54-937](https://github.com/Automattic/wp-calypso/assets/115007291/48c923c2-0a54-448b-b769-c9a224ad3699)

## Proposed Changes

* Replace current search component used within the Search bar of Tools > Advertising with the one used in https://wordpress.com/sites:

The current look is
<img width="447" alt="image" src="https://github.com/Automattic/wp-calypso/assets/115007291/f735a5af-de42-4455-9b5a-e30ce4710b12">

The requested look is:
<img width="442" alt="image" src="https://github.com/Automattic/wp-calypso/assets/115007291/9bd02b52-deda-4faf-9317-d3fe6c1e5187">

* Fix breaking a page when `#` is used as a search term:

https://github.com/Automattic/wp-calypso/assets/115007291/eb998506-b201-43d1-8492-846e748cf071

## Testing Instructions



* Open Tools > Advertising
* Check that magnifier and cross icons are looks like in https://wordpress.com/sites
* Check that the `font-size`/`line-height` are still `14px`/`20px`
* Try using `#` as a search term for both posts and campaigns, it should turned out into "No posts/campaigns found" kind of response (the app should not break):

https://github.com/Automattic/wp-calypso/assets/115007291/3a4fb2b5-6c31-443e-8743-e0075848841e

## Pre-merge Checklist
